### ISSUE TITLE
[Bug] `exclude` option not working

### ISSUE THREAD

**Issue by FatehAK** [jw-12138/vite-plugin-vsharp#7]
The option to exclude certain files does not seem to work. I checked the source code it seems you are splicing the entire array and removing all the entries by `thisName.splice(0, 2);` as a result thisName ends up being an empty string.

Also, lets say I give exclude like so -
`exclude: ['thumbsup.gif']`

When running `vite build` the filename contains the hash as well so the output of `_path["default"].basename(el).split('.')` would be thumbsup-[hash].gif

This will fail the comparison check `opts.exclude.includes(thisName + thisExt)`

**Comment by jw-12138** [comment]
@FatehAK could you please provide the file/folder structure of your project?

**Comment by FatehAK** [comment]
It looks like this:

![image](https://user-images.githubusercontent.com/34565559/207002178-45cd9eec-1052-409e-8ae1-de4605e2b240.png)

**Comment by jw-12138** [comment]
Still can't reproduce this, if you are using the image directly (using `<img>` tag) inside `index.html` or related components, the `exclude` option should work as expected. 

File hash generated by Vite should have no impact with comparison check. 

could you please provide more details or a sample project?

**Comment by FatehAK** [comment]
I'm using an `<img>` tag inside index.html. You can take a look at this repo - 

https://github.com/FatehAK/memrise/tree/vsharp-test

**Comment by jw-12138** [comment]
Okay, I think I know where the problem is...  

Vite 4 bundles things differently than Vite 2/3, The filename will be `<filename>-<hash>.<ext>` instead of `<filename>.<hash>.<ext>`.

for a temporary solution, you can downgrade vite to `3.2.5` from `4.0.0`, just run `npm i vite@3`. 

I'll see if I could do something to round this up!

**Comment by FatehAK** [comment]
Nice catch! No wonder the slicing logic worked fine you would have had 3 elements in the array earlier.

**Comment by jw-12138** [comment]
I found a solution in https://github.com/vitejs/vite/issues/378, basically you can add something in your `vite.config.js` like this if you still want to use Vite 4:

```javascript
import vsharp from "vite-plugin-vsharp"
export default ({
  // ⬇️ from here
  build: {
    rollupOptions: {
      output: {
        assetFileNames: `assets/[name].[hash].[ext]`
      }
    }
  },
  // ⬆️ to here
  plugins: [
    vsharp()
  ]
})
```

not very elegant, but it works. 🥲

**Comment by FatehAK** [comment]
https://rollupjs.org/guide/en/#outputassetfilenames

Seems like Vite 4.0 upgraded to Rollup 3.0 which has changed the default asset names.

Is it possible to handle this within the vsharp plugin itself? Since everyone would have to make the path change to `assets/[name].[hash].[ext]` 😕

**Comment by jw-12138** [comment]
This is now fixed at version `>1.4`, you can upgrade the vsharp plugin and delete the config items. 😆
